surface_potential_analysis/kernel/build.py

```diff
@@ -0,0 +1,172 @@
+        [np.ndarray[Any, np.dtype[np.float64]]],
+        np.ndarray[Any, np.dtype[np.complex128]],
+    ],
+) -> IsotropicNoiseKernel[
+    TupleBasisWithLengthLike[*tuple[FundamentalPositionBasis[Any, Any], ...]],
+]:
+    """
+    Get an Isotropic Kernel with a correllation beta(x-x').
+
+    Parameters
+    ----------
+    basis : StackedBasisWithVolumeLike[Any, Any, Any]
+    fn : Callable[
+        [np.ndarray[Any, np.dtype[np.float64]]],
+        np.ndarray[Any, np.dtype[np.complex128]],
+    ]
+        beta(x-x'), the correllation as a function of displacement
+
+    Returns
+    -------
+    IsotropicNoiseKernel[
+        TupleBasisWithLengthLike[*tuple[FundamentalPositionBasis[Any, Any], ...]],
+    ]
+    """
+    displacements = get_displacements_x(basis)[0]
+    correlation = fn(displacements)
+
+    basis_x = stacked_basis_as_fundamental_position_basis(basis)
+    return {"basis": basis_x, "data": correlation.ravel()}
+
+
+def _get_temperature_corrected_operators(
+    hamiltonian: SingleBasisOperator[_B1],
+    operators: SingleBasisOperatorList[
+        _B0,
+        _B2,
+    ],
+    temperature: float,
+) -> SingleBasisOperatorList[
+    _B0,
+    _B1,
+]:
+    commutator = get_commutator_operator_list(hamiltonian, operators)
+    correction = scale_operator_list(-1 / (4 * Boltzmann * temperature), commutator)
+    return add_list_list(correction, operators)
+
+
+def get_temperature_corrected_noise_operators(
+    hamiltonian: SingleBasisOperator[_B1],
+    operators: SingleBasisNoiseOperatorList[
+        _B0,
+        _B2,
+    ],
+    temperature: float,
+) -> SingleBasisNoiseOperatorList[
+    _B0,
+    _B1,
+]:
+    """
+    Get the noise operators, applying the caldeira-legget like temperature correction.
+
+    Parameters
+    ----------
+    hamiltonian : SingleBasisOperator[_B1]
+    operators : SingleBasisDiagonalNoiseOperatorList[ _B0, _B1, ]
+    temperature : float
+
+    Returns
+    -------
+    SingleBasisNoiseOperatorList[ _B0, _B1]
+    """
+    corrected_operators = _get_temperature_corrected_operators(
+        hamiltonian,
+        operators,
+        temperature,
+    )
+
+    return {
+        "basis": corrected_operators["basis"],
+        "data": corrected_operators["data"],
+        "eigenvalue": operators["eigenvalue"],
+    }
+
+
+def get_temperature_corrected_diagonal_noise_operators(
+    hamiltonian: SingleBasisOperator[_B1],
+    operators: SingleBasisDiagonalNoiseOperatorList[
+        _B0,
+        _B2,
+    ],
+    temperature: float,
+) -> SingleBasisNoiseOperatorList[
+    _B0,
+    _B1,
+]:
+    """
+    Get the noise operators, applying the caldeira-legget like temperature correction.
+
+    Parameters
+    ----------
+    hamiltonian : SingleBasisOperator[_B1]
+    operators : SingleBasisDiagonalNoiseOperatorList[ _B0, _B1, ]
+    temperature : float
+
+    Returns
+    -------
+    SingleBasisNoiseOperatorList[ _B0, _B1]
+    """
+    operators_full = as_noise_operators(operators)
```

Comment:
if this is get temperature corrected diagonal noise op, why is as_diagonal_noise_op not used instead?

Reply:
Good point - that is a bit wasteful